File: src/main/java/alfheim/common/block/BlockDreamLog.java
package alfheim.common.block;

import alfheim.AlfheimCore;
import alfheim.api.ModInfo;
import alfheim.common.lexicon.AlfheimLexiconData;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.block.BlockLog;
import net.minecraft.client.renderer.texture.IIconRegister;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.IIcon;
import net.minecraft.world.World;
import vazkii.botania.api.lexicon.ILexiconable;
import vazkii.botania.api.lexicon.LexiconEntry;

public class BlockDreamLog extends BlockLog implements ILexiconable {

	public IIcon[] textures = new IIcon[2]; 
	
	public BlockDreamLog() {
		super();
		this.setBlockName("DreamLog");
		this.setCreativeTab(AlfheimCore.alfheimTab);
	}

	@SideOnly(Side.CLIENT)
    public void registerBlockIcons(IIconRegister reg) {
		textures[0] = reg.registerIcon(ModInfo.MODID + ":DreamLogTop");
		textures[1] = reg.registerIcon(ModInfo.MODID + ":DreamLogSide");
    }
	
	@SideOnly(Side.CLIENT)
	protected IIcon getSideIcon(int meta) {
		return textures[1];
	}

	@SideOnly(Side.CLIENT)
	protected IIcon getTopIcon(int meta) {
		return textures[0];
	}

	@Override
	public LexiconEntry getEntry(World world, int x, int y, int z, EntityPlayer player, ItemStack lexicon) {
		return AlfheimLexiconData.worldgen;
	}
}

File: src/main/java/alfheim/common/block/BlockDreamLog.kt
package alfheim.common.block

import alfheim.common.block.base.BlockModRotatedPillar
import alfheim.common.item.block.ItemBlockMod
import alfheim.common.lexicon.AlfheimLexiconData
import cpw.mods.fml.common.registry.GameRegistry
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.world.*
import vazkii.botania.api.lexicon.ILexiconable

class BlockDreamLog: BlockModRotatedPillar(Material.wood), ILexiconable {
	
	init {
		setBlockName("DreamLog")
		blockHardness = 2f
	}
	
	override fun register(name: String) {
		GameRegistry.registerBlock(this, ItemBlockMod::class.java, name)
	}
	
	override fun breakBlock(world: World, x: Int, y: Int, z: Int, block: Block, fortune: Int) {
		val b0: Byte = 4
		val i1: Int = b0 + 1
		
		if (world.checkChunksExist(x - i1, y - i1, z - i1, x + i1, y + i1, z + i1)) {
			for (j1 in -b0..b0) for (k1 in -b0..b0)
				for (l1 in -b0..b0) {
					val blockInWorld: Block = world.getBlock(x + j1, y + k1, z + l1)
					if (blockInWorld.isLeaves(world, x + j1, y + k1, z + l1)) {
						blockInWorld.beginLeavesDecay(world, x + j1, y + k1, z + l1)
					}
				}
		}
		super.breakBlock(world, x, y, z, block, fortune)
	}
	
	override fun canSustainLeaves(world: IBlockAccess, x: Int, y: Int, z: Int) = true
	override fun isWood(world: IBlockAccess?, x: Int, y: Int, z: Int) = true
	override fun getEntry(world: World, x: Int, y: Int, z: Int, player: EntityPlayer, lexicon: ItemStack) = AlfheimLexiconData.worldgen
}
